frontend/integration-tests/tests/deploy-image.scenario.ts

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable no-undef, no-unused-vars */
+
+import { browser, $, element, by, ExpectedConditions as until, Key } from 'protractor';
+import * as _ from 'lodash';
+
+import { appHost, checkLogs, checkErrors, testName } from '../protractor.conf';
+import * as crudView from '../views/crud.view';
+
+describe('Deploy Image', () => {
+  afterEach(() => {
+    checkLogs();
+    checkErrors();
+  });
+
+  describe('Deployment Configs page', () => {
+    it('has Create button', async() => {
+      await browser.get(`${appHost}/k8s/ns/${testName}/deploymentconfigs`);
+      await crudView.isLoaded();
+      expect((element(by.buttonText('Create'))).isPresent()).toBe(true);
+    });
+
+    it('can be used to navigate to the Deploy Image page', async() => {
+      await $('#item-create').click().then(() => browser.actions().sendKeys(Key.ARROW_DOWN, Key.ENTER).perform());
+      await browser.wait(until.presenceOf($('.co-deploy-image')));
+      expect((element(by.cssContainingText('#resource-title', 'Deploy Image'))).isPresent()).toBe(true);
+    });
+  });
+
+  describe('Deploy Image page', () => {
+    it('can be used to search for an image', async() => {
+      await $('#imageName').sendKeys('mysql');
+      await $('.input-group-btn .btn-default').click();
+      await browser.wait(until.presenceOf($('.co-image-name-results__details')));
+      expect((element(by.cssContainingText('.co-image-name-results__heading', 'mysql'))).isPresent()).toBe(true);
+    });
+
+    it('can be used to create an app based on an image', async() => {
+      await $('.co-m-btn-bar .btn-primary').click();
+      await browser.wait(until.presenceOf($('.co-m-DeploymentConfig')));
```

Comment:
This class name convention doesn't feel right here. Both BEM and usage seems wrong.
Also can't find any of it's occurrence.

Reply:
Agree it's strange.  Is coming from https://github.com/openshift/console/blob/master/frontend/public/components/factory/details.tsx#L35.

Reply:
I'd suggest we add something to crud view if it's not there to wait until a details page is loaded. There are other places we need it. Then we can consistent checks in all our tests.